src/plugins/WebOfTrust/util/TickerDelayedBackgroundJob.java

```diff
@@ -0,0 +1,363 @@
+package plugins.WebOfTrust.util;
+
+import freenet.node.FastRunnable;
+import freenet.node.PrioRunnable;
+import freenet.support.Executor;
+import freenet.support.Ticker;
+import freenet.support.io.NativeThread;
+
+/**
+ * A {@link DelayedBackgroundJob} implementation that uses a {@link Ticker} for scheduling.
+ *
+ * @author bertm
+ * @see TickerDelayedBackgroundJobFactory
+ */
+public class TickerDelayedBackgroundJob implements DelayedBackgroundJob {
+    static enum JobState {
+        /** Waiting for a trigger, no running job thread or scheduled job. */
+        IDLE,
+        /** Waiting for the scheduled job to be executed, no running job thread. */
+        WAITING,
+        /** Running the job in the job thread. */
+        RUNNING,
+        /** Waiting for the running job thread to finish before we terminate. */
+        TERMINATING,
+        /** Terminated, no running job thread. */
+        TERMINATED
+    }
+
+    /** Constant for {@link #nextExecutionTime} meaning there is no next execution requested. */
+    private final static long NO_EXECUTION = Long.MAX_VALUE;
+
+    /** Job wrapper for status tracking. */
+    private final DelayedBackgroundRunnable realJob;
+    /** Human-readable name of this job. */
+    private final String name;
+    /** Aggregation delay in milliseconds. */
+    private final long defaultDelay;
+    /** The ticker used to schedule this job. */
+    private final Ticker ticker;
+    /** The executor of this background job. */
+    private final Executor executor;
+
+    /** Tiny job to run on the ticker, invoking the execution of the {@link #realJob} on the
+     * executor (only set in {@code WAITING} state). */
+    private Runnable waitingTickerJob = null;
+    /** Running state of this job. */
+    private JobState state = JobState.IDLE;
+    /** Job execution thread, only if the job is running (only set in {@code RUNNING} state). */
+    private Thread thread = null;
+    /** Next absolute time we have a job execution scheduled, or {@link #NO_EXECUTION} if none. */
+    private long nextExecutionTime = NO_EXECUTION;
+
+    /**
+     * Constructs a delayed background job with the given default delay. Negative delays
+     * are treated as zero delay.
+     * The {@link Ticker} given and its {@link Executor} <b>must</b> have an asynchronous
```

Comment:
Not a blocker, but an idea for further fred work in case you're interested:
The Ticker interface doesn't have ANY JavaDoc at all, which is part of the reason why you have to specify requirements here.
Given that you're familiar with the stuff now, it might make sense for you to determine the common attributes of all existing Ticker implementations, and add JavaDoc to the Ticker interface which specifies them.
Especially the pitfalls which caused this pull request at first should be documented: Remember, I had wrongly assumed that TrivialTicker will never execute the same job twice in parallel if I set "no duplicates = true".
This would also make validating things which use Ticker such as this pull request a lot easier for others in the future - right now I have to dig into the Ticker implementation source code to understand whether what you're doing is correct.

(Didn't check the situation of Executor, maybe the above also applies to that)